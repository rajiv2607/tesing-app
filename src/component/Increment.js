import React, { useState } from 'react';

function App() {
  let [count, setCount] =  useState(0)


  const handleIncrement = () => {
    setCount(count + 1);
  };


  function  handleDecrement(){
    setCount(count - 1);
  }

  return (
    <div className="App">
     <h1>Testing </h1>
     <button className='decrement' onClick={handleDecrement}>-</button>
     <label className='count'>{count}</label>
     <button className='increment' onClick={handleIncrement} >+</button>
    </div>
  );
}

export default App;
