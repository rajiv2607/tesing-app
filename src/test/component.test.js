import Increment from '../component/Increment'
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('my first test' , () =>{
    it('this is one' , () =>{
        expect(true)
    });

    it('find the number of elements present', () => {
        const increment = shallow(<Increment />);
        expect(increment.find('button').length).toBe(2);
        expect(increment.find('label').length).toBe(1);
    });

    it('get the initial state of count', () => {
        const increment = shallow(<Increment />);
       expect(increment.find('.count').text()).toBe("0");
    });

    it('get the state after we click the increment', () => {
        const increment = shallow(<Increment />);
       increment.find('.increment').simulate('click')
       expect(increment.find('.count').text()).toBe("1");
    });

    it('get the state after we click the decrement', () => {
        const increment = shallow(<Increment />);
       increment.find('.decrement').simulate('click')
       expect(increment.find('.count').text()).toBe("-1");
    });

   
})

// Snapshot testing 
describe('Increment component', () => {
    it('should render snapshot', () => {
      const component = renderer.create(<Increment />);
  
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })
})