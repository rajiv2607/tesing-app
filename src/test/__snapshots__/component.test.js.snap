// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Increment component should render snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Testing 
  </h1>
  <button
    className="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <label
    className="count"
  >
    0
  </label>
  <button
    className="increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
